migrator.go

```diff
@@ -298,8 +304,9 @@ func (m Migrator) ColumnTypes(value interface{}) (columnTypes []gorm.ColumnType,
 		currentSchema, table := m.CurrentSchema(stmt, stmt.Table)
 		columns, err := m.DB.Raw(
 			"SELECT column_name, is_nullable, udt_name, character_maximum_length, "+
-				"numeric_precision, numeric_precision_radix, numeric_scale, datetime_precision "+
-				"FROM information_schema.columns WHERE table_catalog = ? AND table_schema = ? AND table_name = ?",
+				"numeric_precision, numeric_precision_radix, numeric_scale, datetime_precision, 8 * typlen "+
```

Comment:
do we need the `8 * typlen`?

Reply:
From looking at the code, the value produced by `migrator.ColumnType.Length()` is expected to be in bits, so we either need to multiply by 8 here or in the `Length()` accessor. I thought doing it in the SQL expression would be less confusing.

Reply:
thank you for your pr.